docs/library/machine.Pin.rst

```diff
@@ -3,16 +3,42 @@
 class Pin -- control I/O pins
 =============================
 
-A pin is the basic object to control I/O pins (also known as GPIO -
-general-purpose input/output). It has methods to set
-the mode of the pin (input, output, etc) and methods to get and set the
-digital logic level. For analog control of a pin, see the ADC class.
+A pin object is used to control I/O pins (also known as GPIO - general-purpose
+input/output).  Pin objects are commonly associated with a physical pin that can
+drive a voltage and read voltages.  The pin class has methods to set the mode of
+the pin (IN, OUT, etc) and methods to get and set the digital logic level.
+For analog control of a pin, see the ADC class.
 
-Usage Model:
+A specific pin is constructed by using its identifier, which is specific to a
+given port.  Identifiers can always be integers and on some ports strings can be
+used to name a pin.
+
+Usage Model::
+
+    from machine import Pin
+
+    # create an output pin on pin #0
+    p0 = Pin(0, Pin.OUT)
+
+    # set the value low then high
+    p0.value(0)
+    p0.value(1)
+
+    # create an input pin on pin #2, with a pull up resistor
+    p2 = Pin(2, Pin.IN, Pin.PULL_UP)
+
+    # read and print the pin value
+    print(p2.value())
+
+    # reconfigure pin #0 in input mode
+    p0.mode(p0.IN)
+
+    # configure an irq callback
+    p0.irq(lambda p:print(p))
 
 .. only:: port_wipy
```

Comment:
Ok, so we definitely need to get rid of "only" stuff. 2 approaches: 1) make the main doc page like this be truly generic and abstract, and have port-specific info under port reference. Easier, but much less helpful to users. 2) Say that for "finite number of officially supported ports, we have such information inline - always visible to each user".
@deshipu